ksqldb-engine/src/test/java/io/confluent/ksql/ddl/commands/RegisterTypeFactoryTest.java

```diff
@@ -17,33 +17,99 @@
 
 import static org.hamcrest.CoreMatchers.equalTo;
 import static org.hamcrest.MatcherAssert.assertThat;
+import static org.junit.Assert.assertThrows;
+import static org.mockito.Mockito.when;
 
 import io.confluent.ksql.execution.ddl.commands.RegisterTypeCommand;
 import io.confluent.ksql.execution.expression.tree.Type;
+import io.confluent.ksql.metastore.MetaStore;
 import io.confluent.ksql.parser.tree.RegisterType;
 import io.confluent.ksql.schema.ksql.types.SqlBaseType;
 import io.confluent.ksql.schema.ksql.types.SqlPrimitiveType;
 import io.confluent.ksql.schema.ksql.types.SqlStruct;
+import io.confluent.ksql.schema.ksql.types.SqlType;
+import io.confluent.ksql.util.KsqlException;
 import java.util.Optional;
+import org.junit.Before;
 import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
 
+@RunWith(MockitoJUnitRunner.class)
 public class RegisterTypeFactoryTest {
-  private final RegisterTypeFactory factory = new RegisterTypeFactory();
+  private static final String EXISTING_TYPE = "existing_type";
+  private static final String NOT_EXISTING_TYPE = "not_existing_type";
+  private RegisterTypeFactory factory;
 
-  @Test
+  @Mock
+  private MetaStore metaStore;
+  @Mock
+  private SqlType customType;
+
+  @Before
+  public void setUp() {
+    when(metaStore.resolveType(EXISTING_TYPE)).thenReturn(Optional.of(customType));
+    factory = new RegisterTypeFactory(metaStore);
+  }
+
+    @Test
   public void shouldCreateCommandForRegisterType() {
```

Comment:
Add another test case to create a non-existing type when ifNotExists is not set. This should be used to validate we won't get errors in such scenario.